dal/mcd_validate.py

```diff
@@ -0,0 +1,281 @@
+import datetime
+import enum
+import math
+from dataclasses import dataclass
+from typing import Dict, Callable, Optional, List
+
+import pytz
+
+from .mcd_model import FCState
+
+# the purpose of this file is to provide you with a common validators for MCD database
+
+class FieldType(enum.Enum):
+    FLOAT = 1
+    INT = 2
+    TEXT = 3
+    COMMENT_THREAD = 4
+    ISO_DATE = 5
+
+
+class DeviceType(enum.Enum):
+    """Defines what kind of device we are dealing with, so we can use the right validator for checking data before
+       inserting it into db.
+
+       NOTE: this is an append only enumeration. If you deprecate a device type, this device type enum
+       should NEVER BE reused, or you will end up fetching invalid device type from history and validation will
+       report it as a broken device when in fact was perfectly fine at the time of insertion.
+    """
+    # Alternatively we could use text instead of ints (which makes db easier to inspect at the cost of additional space)
+    # NOTE: the reason why we have Unset and Unknown device is to do the right thing during validation.
+    #   0 - type usually means somebody forgot to set a type (which is a bug in MCD 2.0)
+    #   1 - type means somebody set the type to Unknown on purpose and device data will not be validated
+    Unset = 0
+    Unknown = 1
+    Mcd = 2
+    Source = 3
+    Blank = 4
+    Aperture = 5
+    FlatMirror = 6
+    KbMirror = 7
+    Crl = 8
+    Crystal = 9
+    Grating = 10
+
+
+@dataclass
```

Comment:
Linking to a comment you made further down about `FieldValidator`'s being immutable, we could codify that by actually freezing the dataclass instances 
```suggestion
@dataclass(frozen=True)
```